examples/uart-dma-rx.rs

```diff
@@ -0,0 +1,94 @@
+#![deny(warnings)]
+#![deny(unsafe_code)]
+#![no_main]
+#![no_std]
+
+extern crate cortex_m_rt as rt;
+
+use hal::dma::{config::DmaConfig, stream::DMAExt, TransferExt};
+use hal::prelude::*;
+use hal::pwr::PwrExt;
+use hal::serial::*;
+use hal::{rcc, stm32};
+use stm32g4xx_hal as hal;
+
+use cortex_m_rt::entry;
+use utils::logger::info;
+
+#[macro_use]
+mod utils;
+
+#[entry]
+fn main() -> ! {
+    utils::logger::init();
+
+    info!("Initializing...");
+
+    let dp = stm32::Peripherals::take().expect("cannot take peripherals");
+
+    let rcc = dp.RCC.constrain();
+    let pwr = dp.PWR.constrain().freeze();
+    let mut rcc = rcc.freeze(rcc::Config::hsi(), pwr);
+
+    let streams = dp.DMA1.split(&rcc);
+    let config = DmaConfig::default()
+        .transfer_complete_interrupt(false)
+        .circular_buffer(true)
+        .memory_increment(true);
+
+    info!("Init UART");
+    //let gpioa = dp.GPIOA.split(&mut rcc);
+    //let tx = gpioa.pa2.into_alternate();
+    //let rx = gpioa.pa3.into_alternate();
+    let gpioc = dp.GPIOC.split(&mut rcc);
+    let tx = gpioc.pc10.into_alternate();
+    let rx = gpioc.pc11.into_alternate();
```

Comment:
Anyone have any preference for what pins to use? The commented out lines are same as `uart-dma` example